### DISCUSSION TITLE
Help with AptosCoin Transfer from Contract in Move

### DISCUSSION THREAD

**Discussion by Hossein-79** [aptos-labs/aptos-developer-discussions#458] [Questions] [answered]
### Discord user ID

hossein_79

### Describe your question in detail.

I'm working on a smart contract in Move where I need to handle payments and transfers for a cause-based system. The setup involves users buying tickets for a cause, and when the cause is closed, a portion of the funds (a charity percentage) should go to the cause creator, while the remaining amount should be transferred to the winner.

However, I understand that Move contracts themselves cannot hold funds directly. What’s the best approach in Aptos to simulate this kind of logic, where the funds need to be collected and then distributed to different addresses after an event (like closing a cause)?

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by ajaythxkur** [comment, marked as answer]
@Hossein-79  I had asked same question once
[Discussion](https://github.com/aptos-labs/aptos-developer-discussions/discussions/407#discussioncomment-10609705)
